### PR TITLE
Anagram of String in C++

### PR DESCRIPTION
**Pull request by sukritidawar**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Added the solution in cp folder.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by sukritidawar** [comment]
@Kajol-Kumari @plazzy99 please review.

File: C-Plus-Plus/cp/check_anagrams.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    cout << "Anagrams or not: ";
+    if(check_anagrams(s1, s2)){
+        cout << "yes";
+    }
+    else{
+        cout << "no";
+    }
+    return 0;
+}
+// Dry run:
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/cp/check_anagrams.cpp]: remove dry run and add a sample test case

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/check_anagrams.cpp (hunk 1 above)
Comment (sukritidawar) [reply C-Plus-Plus/cp/check_anagrams.cpp]: done

**Review by vatsalkesarwani12 — APPROVED** [review]

File: C-Plus-Plus/cp/check_anagrams.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            return false; 
+        }
+    }
+    return true;                       //If all elements of count array are zero, s1 and s2 are anagrams
+}
+
+int main(){
+    string s1, s2;
+    cout << "Enter s1";
+    getline(cin, s1);
```
Comment (atarax665) [inline C-Plus-Plus/cp/check_anagrams.cpp]: We can even use normal cin to take string input.

File: C-Plus-Plus/cp/check_anagrams.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        if(count[i] != 0){              //If two strings are anagrams, all the count elements should be zero, and if they are not zero, we return false
+            return false; 
+        }
+    }
+    return true;                       //If all elements of count array are zero, s1 and s2 are anagrams
+}
+
+int main(){
+    string s1, s2;
+    cout << "Enter s1";
```
Comment (atarax665) [inline C-Plus-Plus/cp/check_anagrams.cpp]:
```suggestion
    cout << "Enter s1: ";
```

File: C-Plus-Plus/cp/check_anagrams.cpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    getline(cin, s1);
+    cout << "Enter s2";
```
Comment (atarax665) [inline C-Plus-Plus/cp/check_anagrams.cpp]:
```suggestion
    cout << "Enter s2: ";
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/check_anagrams.cpp (hunk 2 above)
Comment (sukritidawar) [reply C-Plus-Plus/cp/check_anagrams.cpp]: done

File: C-Plus-Plus/cp/check_anagrams.cpp (hunk 3 above)
Comment (sukritidawar) [reply C-Plus-Plus/cp/check_anagrams.cpp]: done

File: C-Plus-Plus/cp/check_anagrams.cpp (hunk 4 above)
Comment (sukritidawar) [reply C-Plus-Plus/cp/check_anagrams.cpp]: done

**Review by atarax665 — APPROVED** [review]
LGTM :+1:

**Review by Kajol-Kumari — APPROVED** [review]
well commented and organised! LGTM :)

**Pull request merged**